### PR TITLE
Add a tutorial on serving custom Error 404 page

### PR DESCRIPTION
**Pull request by ashmaroli**
/cc @jekyll/documentation

### CODE REVIEW

File: docs/_tutorials/custom-404-page.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+Add the following to the ngnix configuration file, `nginx.conf`, which is usually located inside `/etc/nginx/` or `/etc/nginx/conf/`:
+
+```
+server {
+  error_page 404 /404.html;
+  location  /404.html {
+    internal;
+  }
+}
+```
```
Comment (ashmaroli) [inline docs/_tutorials/custom-404-page.md]: @crispgm Since I don't have ready access to a nginx server, can you please test the above? Thanks

Comment (crispgm) [inline docs/_tutorials/custom-404-page.md]: Typically, the configuration is like:

```
server {
    listen 80; # listen 443 ssl for https site
    location / {
         root   /path/to/your/site;
         index  index.html;
         error_page 404 /404.html;
    }
}
```

Comment (ashmaroli) [inline docs/_tutorials/custom-404-page.md]: okay, are all those declarations within the scope of this tutorial?

**Comment by DirtyF** [comment]
Why do we need this ? This information is partly already available: https://help.github.com/articles/creating-a-custom-404-page-for-your-github-pages-site/

What do you think @tomjohnson1492 ?

**Comment by ashmaroli** [comment]
> Why do we need this ?

A [request](https://github.com/jekyll/jekyll/pull/5945#issuecomment-285842244) from Pat Hawks
Another [request](https://github.com/jekyll/jekyll/pull/5945#issuecomment-285845638) from David Zhang

**Comment by tomjohnson1492** [comment]
I think this tutorial is a worthwhile addition to the Jekyll docs. Although some of the content addresses 404 pages generally and has overlap with some github doc info, the first part of this tutorial is more focused on Jekyll and is something that many people may need. I think this would be a good addition to the tutorials section.

**Comment by DirtyF** [comment]
@jekyllbot: merge +docs

**Pull request merged**